BullsAndCowsTest/BullsAndCowsGameTest.cs

```diff
@@ -8,10 +9,113 @@ public class BullsAndCowsGameTest
         [Fact]
         public void Should_create_BullsAndCowsGame()
         {
-            var secretGenerator = new SecretGenerator();
+            var secretGenerator = new TestSecretGenerator();
             var game = new BullsAndCowsGame(secretGenerator);
             Assert.NotNull(game);
             Assert.True(game.CanContinue);
         }
+
+        [Fact]
+        public void Should_terminate_game_after_6_guesses()
+        {
+            var secretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(secretGenerator);
+            for (int i = 0; i < 5; i++)
+            {
+                game.Guess("5 2 3 4");
+            }
+
+            Assert.True(game.CanContinue);
+
+            game.Guess("5 2 3 4");
+            Assert.False(game.CanContinue);
+        }
+
+        [Fact]
+        public void Should_return_0A0B_when_all_wrong()
+        {
+            //given
+            var secretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(secretGenerator);
+            //when
+            string answer = game.Guess("5 6 7 8");
+            //then
+            Assert.Equal("0A0B", answer.Split(" ")[0]);
```

Comment:
Why using answer.Split(" ")[0]?

Reply:
@luhouzi12 my "answer" contains bulls/cows and additional message, separated by white space. This is a good point and I will rename the variable to eliminate ambiguity.